chain/src/address.rs

```diff
@@ -1,14 +1,40 @@
-use super::*;
 use schemars::JsonSchema;
```

Comment:
These changes were mostly due to trying to move address to its own module to avoid pulling in `chain` but that isn't as easy to do as it relies on multiple things in `chain`. I kept these small changes as it will make it easier to try and move it out if we ever need to do that again to optimize space or for whatever other reason.

Reply:
will the address be a separate crate in the end?

Reply:
maybe if `chain` becomes more bloated, which could happen once we add in the other eras. The biggest contribution to both is the `crpyto` crate so the extra bit added by `chain` didn't seem significant for crates like `cip36` that needed both addresses + crypto, and addresses need hashes (in `crypto`) anyway. For flint the size differences we're talking about aren't really significant (and we'll need all crates anyway), so it might matter more for dapps that just need 1 part, but for now I don't think we need to worry about it.